source/dub/commandline.d

```diff
@@ -2117,13 +2127,124 @@ class ListCommand : Command {
 		logInfoNoTag("Packages present in the system and known to dub:");
 		foreach (p; dub.packageManager.getPackageIterator()) {
 			if ((pname == "" || pname == p.name) && pvlim.matches(p.version_))
-				logInfo("  %s %s: %s", p.name.color(Mode.bold), p.version_, p.path.toNativeString());
+				logInfoNoTag("  %s %s: %s", p.name.color(Mode.bold), p.version_, p.path.toNativeString());
 		}
 		logInfo("");
 		return 0;
 	}
 }
 
+class SelectCommand : PackageBuildCommand {
+	this() @safe pure nothrow
+	{
+		this.name = "select";
+		this.argumentsPattern = "(no args)|<package> <version|path>|<package> <repository> <commit>";
+		this.description = "Management for dependency selections";
+		this.helpText = [
+			"Shows the full dependency tree if executed without arguments, highlighting "~
+			"dependencies that are conflicting or causing other errors."
```

Comment:
It would be great to have a way to know where the dependency to `<package-name>` comes from.
So e.g. `dub select eventcore` lets me know that my app depends on `vibe-d` which depends on `vibe-core` which depends on `eventcore`.

Reply:
we can move this into requirements for `dub explain`, see #1064